### DISCUSSION TITLE
Lazy VStack OnAppear Send To Action laggy   --  TCA Version - 1.15.0

### DISCUSSION THREAD

**Discussion by Little-tale** [pointfreeco/swift-composable-architecture#3466] [Q&A]
### Description

```swift
LazyVStack {
            ForEach(Array(store.state.videoInfos.enumerated()), id: \.element.id) { index, data in
                MoreCharacterListView(setModel: data)
                    .asButton { // 선택되었을때
                        store.send(.viewEventType(.selectedVideoIndex(index)))
                    }
                    .environmentObject(colorSystem)
                    .padding(.bottom, 10)
                    .onAppear {
                        if index >= store.state.videoInfos.count - 20 {
                            if store.state.listLoadTrigger {
                                store.send(.viewEventType(.videoOnAppear(index)))
                            }
                        }
                    }
            }
        }
        
// feature
        ...
        case let .viewEventType(.videoOnAppear(index)):
                state.listLoadTrigger = false
                return .send(.viewEventType(.sideCheckedIndex(index)))

    case let .viewEventType(.sideCheckedIndex(index)):
                if state.skipIndex < index {
                    state.skipIndex = index
                    
                    if (state.videoInfos.count - 1) - index <= 20 {
                        return fetchVideos(state: &state, isScroll: true)
                            .throttle(id: CancelId.scrollID, for: 2, scheduler: RunLoop.current.eraseToAnyScheduler(), latest: false)
                    }
                }
...

private func fetchVideos(state: inout State, isScroll: Bool) -> Effect<Action> {
        return .run { [state = state] send in
            let start = DispatchTime.now()
            await send(.networkType(.fetchVideos(state.dropDownOptions[state.currentIndex], state.currentStart, state.limit, isScroll: isScroll)))
            let end = DispatchTime.now()
            
            let result = Double(end.uptimeNanoseconds - start.uptimeNanoseconds) / 1_000_000
            
            print("sec", result)
        }
    }
 ```

위와 같은 코드에서
액션을 보내면 렉이 걸립니다.
Scrolling in the same code as above
If you send an action, it will lag.

처음에는 하나의 Action 으로 했었으나 렉이 있어서 액션을 분리하고 ".send" 후 ".run{}" 을 통해 분리하였음에도 문제가 발생합니다.
At first, I did it as one action, but there was a lag, so I separated the action and then separated it through ".send" and then ".run{}", but the problem still occurs.

최소한의 "Action"을 보내기 위해 View에서도 한번만 "Action"을 보내기 위해 조치하였으나 부드럽게 스크롤 되다가 한번씩 끊김 현상이 발생합니다.
View also tried to send "Action" only once in order to send the least "Action", but it scrolls smoothly and then breaks down once in a while.

```swift
LazyVStack {
            ForEach(Array(store.state.videoInfos.enumerated()), id: \.element.id) { index, data in
                MoreCharacterListView(setModel: data)
                    .asButton { // 선택되었을때
                        store.send(.viewEventType(.selectedVideoIndex(index)))
                    }
                    .environmentObject(colorSystem)
                    .padding(.bottom, 10)
                    .task {
                        if index >= store.state.videoInfos.count - 20 {
                            if store.state.listLoadTrigger {
                                store.send(.viewEventType(.videoOnAppear(index)))
                            }
                        }
                    }
            }
        }
```

위와 같이, onAppear -> Task 로 바꾸어도 현상은 동일합니다.
혹시 다른 방법이 있습니까?

As above, the phenomenon is the same even if you switch to onAppear -> Task.
Is there any other way?

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

It's my prediction, but I think it's an issue caused by Reducer operating on MainActor.

### Actual behavior

위와 같습니다.
It's the same as above.

### The Composable Architecture version information

 1.15.0

### Xcode version information

Xcode 16.0

**Comment by stephencelis** [comment]
@Little-tale `LazyVStack` is known to be a performance hit in even vanilla SwiftUI, but compounding it with attaching each element with its own `onAppear`/`task` that mutates observable state definitely sounds problematic, vanilla SwiftUI or otherwise.

Because we don't consider this to be a bug with the library, but rather a matter of library usage and optimization, I'm going to convert to a discussion.

If you'd like, please share a project that reproduces a noticeable issue and folks can try to make specific suggestions on how to work around the issue.

**Reply by Little-tale** [reply]
Is there a way to prevent some of the properties of the structure in the "ObservableState" from updating the view?
